src/modules/UI/components/QRCode/QRCode.ui.js

```diff
@@ -1,21 +1,27 @@
+// @flow
+
 import React, {Component} from 'react'
-import {View} from 'react-native'
-import QrCode from 'react-native-qrcode'
-import platform from '../../../../theme/variables/platform.js'
+import {Animated} from 'react-native'
+import qrcode from 'yaqrcode'
 
 import styles from './styles'
 
-export default class QRCode extends Component {
+type Props = {
+  keyboardUp: boolean,
+  value: string,
+  animationQrSize: {},
+  animationPushUpSize: {}
+}
+
+export default class QRCode extends Component<Props> {
   render () {
     return (
-      <View style={styles.qrCodeBorder}>
-        <QrCode
-          style={styles.qrCode}
-          value={this.props.value}
-          bgColor={styles.qrCodeBackground.color}
-          fgColor={styles.qrCodeForeground.color}
-          size={platform.deviceHeight / 4} />
-      </View>
+      <Animated.View style={[ styles.qrCodeBorder, {marginBottom: this.props.animationPushUpSize}]}>
+        <Animated.Image
+          style={{width: this.props.animationQrSize, height: this.props.animationQrSize}}
+          source={{uri: qrcode(this.props.value)}}
```

Comment:
I think this is causing your slow performance on animation. The qrcode() function might be pretty slow. You should only reprocess the QRcode if the user types in the textfield or if the user entered value changes. During the animation, you should only take the resulting data from qrcode() and animate it. The way you have this coded, you'll call the function qrcode() on every frame which I'm sure will be expensive. Please refactor to fix this performance issue